### PR TITLE
Added compatibilty corresponding to environment parameter for Add-Azu…

### PR DESCRIPTION
**Pull request by Ajay-MS**
…reRmAccount

### CODE REVIEW

File: Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1 [hunk 1]
```diff
@@ -126,8 +126,16 @@ function Initialize-AzureSubscription {
...
             # Else, this is AzureRM.
             try {
-                Write-Host "##[command]Add-AzureRMAccount -ServicePrincipal -Tenant $($Endpoint.Auth.Parameters.TenantId) -Credential $psCredential -EnvironmentName $environmentName"
-                $null = Add-AzureRMAccount -ServicePrincipal -Tenant $Endpoint.Auth.Parameters.TenantId -Credential $psCredential -EnvironmentName $environmentName
+                if(CmdletHasMember -cmdlet "Add-AzureRMAccount" -memberName "EnvironmentName")
+                {
+                    Write-Host "##[command]Add-AzureRMAccount -ServicePrincipal -Tenant $($Endpoint.Auth.Parameters.TenantId) -Credential $psCredential -EnvironmentName $environmentName"
+                    $null = Add-AzureRMAccount -ServicePrincipal -Tenant $Endpoint.Auth.Parameters.TenantId -Credential $psCredential -EnvironmentName $environmentName
+                }
+                else
```
Comment (KrishnaAdityaB) [inline Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1]: is _Environment_ variant latest?

File: Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1 [hunk 2]
```diff
@@ -198,4 +206,20 @@ function Set-UserAgent_Core {
...
+}
+
+function CmdletHasMember {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$cmdlet,
+        [string]$memberName)
+    try{
+        $cmdletParameter = (gcm $cmdlet).Parameters.Keys.Contains($memberName)
```
Comment (KrishnaAdityaB) [inline Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1]: rename this to $hasMember

File: Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1 [hunk 3]
```diff
@@ -198,4 +206,20 @@ function Set-UserAgent_Core {
...
-}
\ No newline at end of file
+}
+
+function CmdletHasMember {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$cmdlet,
+        [string]$memberName)
```
Comment (KrishnaAdityaB) [inline Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1]: why is $memberName not madnatory?

File: Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1 (hunk 1 above)
Comment (Ajay-MS) [inline Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1]: Yes

**Pull request merged**